pkg/apis/configuration/v1/kongingress_types.go

```diff
@@ -51,52 +57,97 @@ type KongIngressList struct {
+	// for transmitting a request to the upstream server.
 	//+kubebuilder:validation:Minimum=0
 	ReadTimeout *int `json:"read_timeout,omitempty" yaml:"read_timeout,omitempty"`
+	// The timeout in milliseconds between two successive write operations
+	// for transmitting a request to the upstream server.
 	//+kubebuilder:validation:Minimum=0
 	WriteTimeout *int `json:"write_timeout,omitempty" yaml:"write_timeout,omitempty"`
 }
 
 // KongIngressRoute contains KongIngress route configuration
 //+ It contains the subset of go-kong.kong.Route fields supported by kongstate.Route.overrideByKongIngress
 type KongIngressRoute struct {
-	Methods                 []*string           `json:"methods,omitempty" yaml:"methods,omitempty"`
-	Headers                 map[string][]string `json:"headers,omitempty" yaml:"headers,omitempty"`
-	Protocols               []*KongProtocol     `json:"protocols,omitempty" yaml:"protocols,omitempty"`
-	RegexPriority           *int                `json:"regex_priority,omitempty" yaml:"regex_priority,omitempty"`
-	StripPath               *bool               `json:"strip_path,omitempty" yaml:"strip_path,omitempty"`
-	PreserveHost            *bool               `json:"preserve_host,omitempty" yaml:"preserve_host,omitempty"`
-	HTTPSRedirectStatusCode *int                `json:"https_redirect_status_code,omitempty" yaml:"https_redirect_status_code,omitempty"`
+	// Methods is a list of HTTP methods that match this Route.
+	Methods []*string `json:"methods,omitempty" yaml:"methods,omitempty"`
+	// Headers contains one or more lists of values indexed by header name
+	// that will cause this Route to match if present in the request.
+	// The Host header cannot be used with this attribute.
+	Headers map[string][]string `json:"headers,omitempty" yaml:"headers,omitempty"`
+	// Protocols is an array of the protocols this Route should allow.
+	Protocols []*KongProtocol `json:"protocols,omitempty" yaml:"protocols,omitempty"`
+	// RegexPriority is a number used to choose which route resolves a given request
+	// when several routes match it using regexes simultaneously.
+	RegexPriority *int `json:"regex_priority,omitempty" yaml:"regex_priority,omitempty"`
+	// StripPath sets When matching a Route via one of the paths
+	// strip the matching prefix from the upstream request URL.
+	StripPath *bool `json:"strip_path,omitempty" yaml:"strip_path,omitempty"`
+	// PreserveHost sets When matching a Route via one of the hosts domain names,
+	// use the request Host header in the upstream request headers.
+	//If set to false, the upstream Host header will be that of the Service’s host.
+	PreserveHost *bool `json:"preserve_host,omitempty" yaml:"preserve_host,omitempty"`
+	// HTTPSRedirectStatusCode is the status code Kong responds with
+	// when all properties of a Route match except the protocol.
+	HTTPSRedirectStatusCode *int `json:"https_redirect_status_code,omitempty" yaml:"https_redirect_status_code,omitempty"`
```

Comment:
I would recommend that we let these attributes breathe a little now that they have documentation for the sake of readability:

```suggestion
	// Methods is a list of HTTP methods that match this Route.
	Methods []*string `json:"methods,omitempty" yaml:"methods,omitempty"`

	// Headers contains one or more lists of values indexed by header name
	// that will cause this Route to match if present in the request.
	// The Host header cannot be used with this attribute.
	Headers map[string][]string `json:"headers,omitempty" yaml:"headers,omitempty"`

	// Protocols is an array of the protocols this Route should allow.
	Protocols []*KongProtocol `json:"protocols,omitempty" yaml:"protocols,omitempty"`

	// RegexPriority is a number used to choose which route resolves a given request
	// when several routes match it using regexes simultaneously.
	RegexPriority *int `json:"regex_priority,omitempty" yaml:"regex_priority,omitempty"`

	// StripPath sets When matching a Route via one of the paths
	// strip the matching prefix from the upstream request URL.
	StripPath *bool `json:"strip_path,omitempty" yaml:"strip_path,omitempty"`

	// PreserveHost sets When matching a Route via one of the hosts domain names,
	// use the request Host header in the upstream request headers.
	// If set to false, the upstream Host header will be that of the Service’s host.
	PreserveHost *bool `json:"preserve_host,omitempty" yaml:"preserve_host,omitempty"`

	// HTTPSRedirectStatusCode is the status code Kong responds with
	// when all properties of a Route match except the protocol.
	HTTPSRedirectStatusCode *int `json:"https_redirect_status_code,omitempty" yaml:"https_redirect_status_code,omitempty"`

```